File: Top_Interview_Questions/20_ValidParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> con;
        for(const auto& i : s) {
            if(i == '(') {con.push(i); continue;}
            if(i == '[') {con.push(i); continue;}
            if(i == '{') {con.push(i); continue;}
            
            if(con.empty()) return false;
            char t = con.top(); con.pop();
            if(i == ')' && t == '(') continue;
            if(i == ']' && t == '[') continue;
            if(i == '}' && t == '{') continue;
            return false;
        }
        return con.empty();
    }
};

File: Question_Java/20_Valid_Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Deque<Character> st = new ArrayDeque<>(s.length());
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '[' || ch == '{') {
                st.push(ch);
            } else if (ch == ')') {
                if (st.isEmpty() || st.peek() != '(') return false;
                st.pop();
            } else if (ch == ']') {
                if (st.isEmpty() || st.peek() != '[') return false;
                st.pop();
            } else if (ch == '}') {
                if (st.isEmpty() || st.peek() != '{') return false;
                st.pop();
            }
        }
        return st.isEmpty();
    }
}